src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -1002,8 +1069,59 @@ SubCmdValidate::execute(const SubCmdOptions& _options) const
     return;
   }
 
-  // -- Now process the subcommand --------------------------------------------
-  // -- process "device" option -----------------------------------------------
+  // -- Process the options --------------------------------------------
+  Report::SchemaVersion schemaVersion = Report::SchemaVersion::unknown;    // Output schema version
+  try {
+    // Output Format
+    schemaVersion = Report::getSchemaDescription(sFormat).schemaVersion;
+    if (schemaVersion == Report::SchemaVersion::unknown) 
+      throw xrt_core::error((boost::format("Unknown output format: '%s'") % sFormat).str());
+
+    // Output file
+    if (!sOutput.empty() && boost::filesystem::exists(sOutput)) 
+        throw xrt_core::error((boost::format("Output file already exists: '%s'") % sOutput).str());
+
+    if (testsToRun.empty()) 
+      throw std::runtime_error("No test given to validate against.");
+
+    // Examine test entries
+    for (const auto &userTestName : testsToRun) {
+      const std::string userTestNameLC = boost::algorithm::to_lower_copy(userTestName);   // Lower case the string entry
+
+      if ((userTestNameLC == "all") && (testsToRun.size() > 1)) 
+        throw xrt_core::error("The 'all' value for the tests to run cannot be used with any other named tests.");
+
+      if ((userTestNameLC == "quick") && (testsToRun.size() > 1)) 
+        throw xrt_core::error("The 'quick' value for the tests to run cannot be used with any other name tests.");
+
+      // Validate all of the test names
+      bool nameFound = false;
+      for (auto &test : testNameDescription) {
+        const std::string testNameLC = boost::algorithm::to_lower_copy(test.first);
+        if (userTestNameLC.compare(testNameLC) == 0) {
+          nameFound = true;
+          break;
+        }
+      }
```

Comment:
Hmm. Does this work for all test names? What if the first test name is valid, but the second one is not? Are we looking for at least one valid test name?

Reply:
Correct, the algorithm will search the _entire collection_ looking for 1 valid test name.  If none found an error will be produced.